src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/AzureAttestationBasedEnclaveProvider.NetCoreApp.cs

```diff
@@ -66,35 +66,37 @@ internal class AzureAttestationEnclaveProvider : EnclaveProviderBase
         #region Public methods
         // When overridden in a derived class, looks up an existing enclave session information in the enclave session cache.
         // If the enclave provider doesn't implement enclave session caching, this method is expected to return null in the sqlEnclaveSession parameter.
-        public override void GetEnclaveSession(string servername, string attestationUrl, out SqlEnclaveSession sqlEnclaveSession, out long counter)
+        public override void GetEnclaveSession(string servername, string attestationUrl, bool generateCustomData, out SqlEnclaveSession sqlEnclaveSession, out long counter, out byte[] customData, out long customDataLength)
         {
-            GetEnclaveSessionHelper(servername, attestationUrl, true, out sqlEnclaveSession, out counter);
+            GetEnclaveSessionHelper(servername, attestationUrl, generateCustomData, out sqlEnclaveSession, out counter, out customData, out customDataLength);
         }
 
         // Gets the information that SqlClient subsequently uses to initiate the process of attesting the enclave and to establish a secure session with the enclave.
-        public override SqlEnclaveAttestationParameters GetAttestationParameters()
+        public override SqlEnclaveAttestationParameters GetAttestationParameters(string attestUrl, byte[] customData, long customDataLength)
```

Comment:
>long [](start = 118, length = 4)

can we ulong

Reply:
Does that mean to change the type of all customDataLength to `ulong`?

Reply:
yes. that would be better. having length as -2 doesn't make any sense

Reply:
Well, I try to compile with `ulong` but I got the following error:

`error CS3001: Argument type 'ulong' is not CLS-compliant`

I find that we have the following configuration which means we can not use unsigned type as parameters passed to public functions. 
`<AssemblyInfo Include="[assembly: System.CLSCompliantAttribute(true)]" />`

Reply:
Change `long` to `int` as per offline discussion.